server/controllers/document.js

```diff
@@ -0,0 +1,212 @@
+          message: 'Offset greater than number of documents or limit param. Cannot fetch',
+          status: '400: Bad request',
+        });
+      } else {
+        res.status(200).send(documents);
+      }
+    });
+  },
+
+// [Restricted] Can only fetch documents of logged in user or public documents
+  getDocumentById: (req, res) => {
+    Document.find({
+      $or: [
+        { $and: [{ _id: req.params.document_id }, { privacy: 'public' }] },
+        { $and: [{ _creatorId: req.decoded.id }, { _id: req.params.document_id }] },
+      ],
+    })
+    .exec((err, documents) => {
+      if (err) {
+        res.send(err);
+      } else if (documents.length === 0) {
+        res.status(401).send({
+          message: 'Cannot access document by that id',
+          status: '401: Unauthorised',
+        });
+      } else {
+        res.status(200).send(documents);
+      }
+    });
+  },
+
+  // [Restricted] Can only edit documents of logged in user
+  updateDocumentById: (req, res) => {
+    Document.findById(req.params.document_id)
+    .where('_creatorId').equals(req.decoded.id)
+    .exec((err, document) => {
+      if (err || document === null) {
+        res.status(404).send({
+          error: err,
+          message: 'Could not update document by the id entered',
+          status: '404: Resource Not Found',
+        });
+      }
+       // Only update if a change has happened
+      if (req.body.title) document.title = req.body.title;
+      if (req.body.content) document.content = req.body.content;
+      if (req.body.privacy) document.privacy = req.body.privacy;
+      if (req.body.sharewith) document.sharewith = req.body.sharewith;
+      if (req.body.access) document.access = req.body.access;
+
+      // Then save the user details
+      document.save(() => {
+        if (err) {
+          res.status(401).send({
+            error: err,
+            message: 'Error saving document',
+            status: '401: Unauthorised',
+          });
+        } else {
+          res.status(200).send({
+            success: true,
+            message: 'Document details updated successfully',
+          });
+        }
+      });
+    });
+  },
+
+// [Restricted] To logged in user
+  deleteDocumentById: (req, res) => {
+    Document.find()
+    .where('_creatorId').equals(req.decoded.id)
+    .exec((err) => {
+      if (err) {
+        res.send(err);
+      } else {
+        Document.remove(
+          {
+            $and: [{ _creatorId: req.decoded.id }, { _id: req.params.document_id }],
+          }
+      , () => {
+          if (err) {
+            res.status(500).send({
+              error: err,
+              message: 'Error deleting document',
+              status: '500: Server Error',
+            });
+          } else {
+            res.status(200).send({
+              success: true,
+              message: 'Document Deleted successfully if document was yours. Failed to delete another users\' documents',
```

Comment:
You could have separated the response to send an error when the user tries to delete a document that isn't theirs.